$OpenBSD$

Make `make test` work.
Remove -Werror* and add these -fno-* CFLAGS here.
Keep tendrils here to make generating future bootstraps easier.

Index: makefile
--- makefile.orig
+++ makefile
@@ -393,7 +393,8 @@ endif
 
 ALLFBCFLAGS += -e -m fbc -w pedantic
 ALLFBLFLAGS += -e -m fbc -w pedantic
-ALLCFLAGS += -Wall -Wextra -Wno-unused-parameter -Werror-implicit-function-declaration
+ALLCFLAGS += -Wall -Wextra -Wno-unused-parameter
+ALLCFLAGS += -fno-exceptions -fno-unwind-tables -fno-asynchronous-unwind-tables
 
 ifeq ($(TARGET_OS),xbox)
   ifeq ($(OPENXDK),)
@@ -712,13 +713,15 @@ help:
 
 ################################################################################
 
-.PHONY: cunit-tests log-tests clean-tests
+.PHONY: test cunit-tests log-tests clean-tests
 
+test: cunit-tests log-tests
+
 cunit-tests:
-	cd tests && make cunit-tests FBC="`pwd`/../bin/fbc -i `pwd`/../inc"
+	cd tests && $(MAKE) cunit-tests FBC="`pwd`/../bin/fbc -i `pwd`/../inc"
 
 log-tests:
-	cd tests && make   log-tests FBC="`pwd`/../bin/fbc -i `pwd`/../inc"
+	cd tests && $(MAKE) log-tests FBC="`pwd`/../bin/fbc -i `pwd`/../inc"
 
 warning-tests:
 	cd tests/warnings && FBC="`pwd`/../../bin/fbc" ./test.sh
@@ -1050,31 +1053,15 @@ FBBOOTSTRAPTITLE := $(FBSOURCETITLE)-bootstrap
 bootstrap-dist:
 	# Precompile fbc sources for various targets
 	rm -rf bootstrap
-	mkdir -p bootstrap/dos
-	mkdir -p bootstrap/linux-x86
-	mkdir -p bootstrap/linux-x86_64
-	mkdir -p bootstrap/win32
-	mkdir -p bootstrap/win64
-	./$(FBC_EXE) src/compiler/*.bas -m fbc -i inc -e -r -v -target dos          && mv src/compiler/*.asm bootstrap/dos
-	./$(FBC_EXE) src/compiler/*.bas -m fbc -i inc -e -r -v -target linux-x86    && mv src/compiler/*.asm bootstrap/linux-x86
-	./$(FBC_EXE) src/compiler/*.bas -m fbc -i inc -e -r -v -target linux-x86_64 && mv src/compiler/*.c   bootstrap/linux-x86_64
-	./$(FBC_EXE) src/compiler/*.bas -m fbc -i inc -e -r -v -target win32        && mv src/compiler/*.asm bootstrap/win32
-	./$(FBC_EXE) src/compiler/*.bas -m fbc -i inc -e -r -v -target win64        && mv src/compiler/*.c   bootstrap/win64
+	mkdir -p bootstrap/openbsd-x86
+	mkdir -p bootstrap/openbsd-x86_64
+	fbc src/compiler/*.bas -m fbc -i inc -e -r -v -target openbsd-x86    && mv src/compiler/*.c bootstrap/openbsd-x86
+	fbc src/compiler/*.bas -m fbc -i inc -e -r -v -target openbsd-x86_64 && mv src/compiler/*.c bootstrap/openbsd-x86_64
 
 	# Ensure to have LFs regardless of host system (LFs will probably on
 	# DOS/Win32, but CRLFs could cause issues on Linux)
-	dos2unix bootstrap/dos/*
-	dos2unix bootstrap/linux-x86/*
-	dos2unix bootstrap/linux-x86_64/*
-	dos2unix bootstrap/win32/*
-	dos2unix bootstrap/win64/*
-
-	# Package FB sources (similar to our "gitdist" command), and add the bootstrap/ directory
-	# Making a .tar.xz should be good enough for now.
-	git -c core.autocrlf=false archive --format tar --prefix "$(FBBOOTSTRAPTITLE)/" HEAD | tar xf -
-	mv bootstrap $(FBBOOTSTRAPTITLE)
-	tar -cJf "$(FBBOOTSTRAPTITLE).tar.xz" "$(FBBOOTSTRAPTITLE)"
-	rm -rf "$(FBBOOTSTRAPTITLE)"
+	dos2unix bootstrap/openbsd-x86/*
+	dos2unix bootstrap/openbsd-x86_64/*
 
 #
 # Build the fbc[.exe] binary from the precompiled sources in the bootstrap/
